public/static/plugins/perf/js/perf.js

```diff
@@ -469,16 +469,41 @@ mciModule.controller('PerfController', function PerfController(
         }
         setTimeout(function(){drawDetailGraph($scope.perfSample, $scope.comparePerfSamples, $scope.task.id)},0);
 
+        // This code loads change points for current task from the mdb cloud
+        var changePointsQ = $q(function(resolve, reject) {
+          stitch.StitchClientFactory.create('evergreen_perf_plugin-wwdoa')
+          .then(function(client) {
+            var db = client.service('mongodb', 'mongodb-atlas').db('perf')
+              return db
+                .collection('change_points')
+                .find({
+                  project: $scope.task.branch,
+                  task: $scope.task.display_name,
+                  variant: $scope.task.build_variant,
+                })
+                .limit(999) // TODO this should prevent some accidents
```

Comment:
What accidents does this prevent? I would expect that this can handle an arbitrary number of documents returned from the db

Reply:
There are 100 samples displayed  on each trend chart (which usually equals to few months of time), however this query matches all change points regardless time period / index. I afraid it could result into huge amount of items, which could affect some part of the app. Moreover, we don't want to overload Atlas cluster with non-efficient work.
There are <<999 change points, so it will not affect the logic.

Reply:
Could you check what the indices are for that collection? I would hope that there is an index on either date or revision order, which will be useful for a sort and then a limit. Currently this query does not guarantee that you get any specific subset of change points